### PR TITLE
feat(pass): tokens

### PR DESCRIPTION
**Pull request by krzysztofdrys**
Add token signing and veryfication to be used by pass.

### CODE REVIEW

File: internal/pass/sign/sign.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	resp, err := s.client.Sign(&kms.SignInput{
+		KeyId:            &s.keyID,
+		Message:          hashedSigningString,
+		MessageType:      &messageType,
+		SigningAlgorithm: aws.String(awsSigningAlgorithm),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to sign the message: %w", err)
+	}
+	keySizeInBytes := 258 / 8
```
Comment (tobiaszheller) [inline internal/pass/sign/sign.go:76]: can you add comment why those numbers? i think it should be 256/8 , Those are taken fromhttps://github.com/matelang/jwt-go-aws-kms/blob/main/jwtkms/kms_signing_method.go#L49?

File: internal/pass/sign/verify.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	"github.com/golang-jwt/jwt/v5"
+)
+
+var ErrInvalidClaims = errors.New("invalid claims")
+
+// CanI establish connection with type tp given claims in token.
+func (s Service) CanI(tokenString string, ct ConnectionType) error {
+	cl := jwt.MapClaims{}
+
+	tokenString = jwtHeader + tokenString
```
Comment (tobiaszheller) [inline internal/pass/sign/verify.go:16]: add comment why

File: internal/pass/sign/verify.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	"errors"
+	"fmt"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+var ErrInvalidClaims = errors.New("invalid claims")
+
+// CanI establish connection with type tp given claims in token.
+func (s Service) CanI(tokenString string, ct ConnectionType) error {
```
Comment (tobiaszheller) [inline internal/pass/sign/verify.go:13]: we want connection_id to be also returned from verify method. That way we don;t need to pass extension_id in each request.

File: internal/pass/sign/sign.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+// SignAndEncode information in the message. The result
+// is second and third part of jwt token. Since the first
+// part is constant it is omitted.
+func (s Service) SignAndEncode(m Message) (string, error) {
+	token := jwt.NewWithClaims(s.signingMethod, jwt.MapClaims{
+		"exp":  m.ExpiresAt.Unix(),
+		"aud":  []string{string(m.ConnectionType)},
+		"c_id": m.ConnectionID,
```
[the remark this answers is not in the document]
Comment (krzysztofdrys) [reply internal/pass/sign/sign.go:29]: Let's leave it for now. I treat this package as POC. Once we start actually using it, we can add something here.

File: tests/pass/kms_test.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+	}
+	t.Log(token)
+	t.Log("Length of the token is", len(token))
+
+	if err := srv.CanI(token, sign.ConnectionTypeBrowserExtensionProxy); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestKeyEncryptionAndVerification(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (krzysztofdrys) [reply tests/pass/kms_test.go:51]: done

File: internal/pass/sign/verify.go (hunk 3 above)
Comment (krzysztofdrys) [reply internal/pass/sign/verify.go:13]: Let's do it once we start "merging" our changes. We will then know, what is actually needed. I think `sign` package is POC right now.

File: internal/pass/sign/verify.go (hunk 2 above)
Comment (krzysztofdrys) [reply internal/pass/sign/verify.go:18]: done

File: internal/pass/sign/sign.go (hunk 1 above)
Comment (krzysztofdrys) [reply internal/pass/sign/sign.go]: done. And you are right about the number.

File: internal/pass/sign/lib_test.go [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+func (e ecdsaSigningMethodWithStaticKey) Sign(signingString string, key interface{}) ([]byte, error) {
+	return jwt.SigningMethodES256.Sign(signingString, e.privateKey)
+}
+
+func (e ecdsaSigningMethodWithStaticKey) Alg() string {
+	return jwt.SigningMethodES256.Alg()
+}
+
+func TestKeyEncryptionAndVerificationHappyPath(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (krzysztofdrys) [reply internal/pass/sign/lib_test.go]: done

File: tests/pass/kms_test.go [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/google/uuid"
+
+	"github.com/twofas/2fas-server/internal/pass/sign"
+)
+
+func TestKeyEncryptionAndVerificationHappyPath(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (krzysztofdrys) [reply tests/pass/kms_test.go]: done

**Review by tobiaszheller — APPROVED** [review]

File: tests/pass/kms_test.go [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+				}
+				serviceWithAnotherKey, err := sign.NewService(*resp.KeyMetadata.KeyId, kmsClient)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				token, err := serviceWithAnotherKey.SignAndEncode(sign.Message{
+					ConnectionID:   uuid.New().String(),
+					ExpiresAt:      now.Add(-time.Hour),
+					ConnectionType: sign.ConnectionTypeBrowserExtensionWait,
```
[the remark this answers is not in the document]
Comment (krzysztofdrys) [reply tests/pass/kms_test.go]: done

**Pull request merged**